src/transformers/models/persimmon/modeling_persimmon.py

```diff
@@ -27,17 +27,169 @@
 from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
 
 from ...activations import ACT2FN
-from ...modeling_outputs import BaseModelOutputWithPast, CausalLMOutputWithPast, SequenceClassifierOutputWithPast
+from ...modeling_outputs import (
+    BaseModelOutputWithPast,
+    CausalLMOutputWithPast,
+    SequenceClassifierOutputWithPast,
+)
 from ...modeling_utils import PreTrainedModel
-from ...utils import add_start_docstrings, add_start_docstrings_to_model_forward, logging, replace_return_docstrings
+from ...utils import (
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    is_flash_attn_2_available,
+    logging,
+    replace_return_docstrings,
+)
 from .configuration_persimmon import PersimmonConfig
 
 
+if is_flash_attn_2_available():
+    from flash_attn import flash_attn_func, flash_attn_varlen_func
+    from flash_attn.bert_padding import index_first_axis, pad_input, unpad_input  # noqa
+
+
 logger = logging.get_logger(__name__)
 
 _CONFIG_FOR_DOC = "PersimmonConfig"
 
 
+# Copied from transformers.models.llama.modeling_llama.AttnMaskConverter
+class AttnMaskConverter:
```

Comment:
instead you can import it from `modeling_attn_mask_utils`:
```python
from ...modeling_attn_mask_utils import AttentionMaskConverter, _prepare_4d_causal_attention_mask
```

Reply:
@younesbelkada 

Made the aforementioned changes but am getting odd errors with `test_flash_attn_2_generate_padding_right` test (all other tests pass).  

For example, here are the outputs with and without FA2:
```python
out=tensor([[38, 17, 72, 39, 27, 98, 50, 53],
        [88, 83, 53, 72, 98, 43, 42, 90],
        [44, 38, 53, 70, 57, 77, 58, 87],
        [61, 57, 26, 71, 68,  5, 76, 60],
        [29,  8, 61, 62, 66, 12, 52, 63],
        [83, 71, 43, 61, 22, 83, 89, 89],
        [50, 38, 11, 22, 42, 13, 65, 14],
        [71, 71,  0,  9,  3, 37, 10, 84],
        [81, 19, 56, 62, 67, 18, 58, 87],
        [39,  9, 49, 48, 22, 13, 72, 20],
        [26, 96,  7, 26, 54, 46, 32,  6],
        [67,  9, 87, 93, 42,  8, 65, 14],
        [25, 83, 70, 30, 32, 92, 25, 64]], device='cuda:0')
out_fa=tensor([[38, 17, 72, 39, 27, 98, 50, 95],
        [88, 83, 53, 72, 98, 43, 42, 30],
        [44, 38, 53, 70, 57, 77, 58, 87],
        [61, 57, 26, 71, 68,  5, 76, 60],
        [29,  8, 61, 62, 66, 12, 52, 28],
        [83, 71, 43, 61, 22, 83, 89, 89],
        [50, 38, 11, 22, 42, 13, 65, 10],
        [71, 71,  0,  9,  3, 37, 10, 84],
        [81, 19, 56, 62, 67, 18, 58, 87],
        [39,  9, 49, 48, 22, 13, 72, 10],
        [26, 96,  7, 26, 54, 46, 32, 94],
        [67,  9, 87, 93, 42,  8, 65, 10],
        [25, 83, 70, 30, 32, 92, 25, 64]], device='cuda:0')
```
Seems like there is some unaccounted for randomness as some sequences match while are others do not.

This is odd given that the `flash_attn_2_inference_padding_{left,right}` tests passes as does `flash_attn_2_generate_left_padding` test.

Any ideas what could be causing this?